src/internal/pachsql/table_info.go

```diff
@@ -82,11 +86,20 @@ func GetTableInfoTx(tx *Tx, tablePath string) (*TableInfo, error) {
 		return nil, errors.EnsureStack(err)
 	}
 	defer rows.Close()
+
+	var cinfos []ColumnInfo
+	var precision, scale sql.NullInt64
 	for rows.Next() {
 		var ci ColumnInfo
-		if err := rows.Scan(&ci.Name, &ci.DataType, &ci.IsNullable); err != nil {
+		if err := rows.Scan(&ci.Name, &ci.DataType, &ci.IsNullable, &precision, &scale); err != nil {
 			return nil, errors.EnsureStack(err)
 		}
+		if precision.Valid {
```

Comment:
Do you even need the `Valid` check here?  If it's false, then the `Int64` is 0, which is what `ci.Precision ` would be if you hadn't assigned to it.